lib/monitor/stats.go

```diff
@@ -11,6 +11,196 @@ import (
 	"strings"
 )
 
+// Stats contains all structures that are related to the computations of stats
+// such as Values (compute the mean/min/max/...), Measurements ( aggregation of
+// Values), Stats (collection of measurements) and DataFilter which is used to
+// apply some filtering before any statistics is done.
+
+// ExtraFields in a RunConfig argument that we may want to parse if present
+var extraFields = [...]string{"bf", "rate", "stampratio"}
+
+// Stats holds the different measurements done
+type Stats struct {
+	// How many peers do we have
+	Peers int
+	// How many peers per machine do we use
+	PPM int // PeerPerMachine
+	// How many machines do we have
+	Machines int
+	// Additionals fields that may appears in the resulting CSV
+	// The additionals fields are created when creating the stats out of a
+	// running config. It will try to read some known fields such as "depth" or
+	// "bf" (branching factor) and add then to its struct
+	Additionals map[string]int
+	addKeys     []string
+	// The measures we have and the keys ordered
+	measures map[string]*Measurement
+	keys     []string
+
+	// ValuesWritten  is to know wether we have already written some values or
+	// not. If yes, we can make sure we dont write new measurements otherwise
+	// the CSV would be garbage
+	valuesWritten bool
+	// The filter used to filter out abberant data
+	filter DataFilter
+}
+
+// Return a NewStats with some fields extracted from the platform run config
+// It enforces the default set of measure to do.
+func NewStats(rc map[string]string) *Stats {
+	s := new(Stats).NewStats()
+	s.readRunConfig(rc)
+	return s
+}
+
+// Read a config file and fills up some fields for Stats struct
+func (s *Stats) readRunConfig(rc map[string]string) {
```

Comment:
Nitpick: Why is it called "readRunConfig" instead of just simply "readConfig"? Moreover, the two dbg.Fatal comments below (lines 59 and 64) speak of a "RunConfig" function but I guess "readRunConfig" is meant?